sjtuthesis/build.lua

```diff
@@ -4,13 +4,32 @@ module           = "sjtuthesis"
 
 sourcefiledir    = "source"
 sourcefiles      = {"sjtuthesis.dtx"}
-installfiles     = {"*.sty"}
+installfiles     = {"*.cls","*.def","*.cfg","*.dtx","*.pdf"}
 
--- docfiledir       = "doc"
+docfiledir       = "doc"
 
 typesetexe       = "xelatex"
+typesetfiles     = {"sjtuthesisuserguide.tex","sjtuthesis.dtx"}
+
 bibtexexe        = "bibtex"
 
 unpackexe        = "xetex"
 unpackopts       = "--interaction=batchmode"
 unpackfiles      = {"sjtuthesis.dtx"}
+
+packtdszip       = true
+
+-- function type_example_latexmk()
```

Comment:
l3build provides two hooks for typesetting document demos:
- `typeset_demo_files()`
- `docinit_hook()`

The name of the hook might need to be changed.

Reply:
I ignored that you used `specialtypesetting` for that.😂

Reply:
I planned to use that, but I found that `latexmk` doesn't work as expected. So I commented them out and use the default `xelatex` instead.

Reply:
Yeah, `latexmk` is not working in `l3build`. Maybe they hate each other.

Reply:
A link for that: https://github.com/latex3/l3build/issues/113

Reply:
It may need to overwrite `typeset()` function to achieve that.🤪